woocommerce/woocommerce-ios
# Store creation M2: domain selector updates

Parts of #8045 


### Description


This PR updated the domain selector screen based on the design feedback and updates in the past two weeks:

- Update placeholder image - dark mode, transparent background
- ~~Auto-focus the text field~~ --> I ended up not doing this because the search is pre-filled with the store name in the previous screen
- Update no search results error
- Update spacing between elements
- Update search field trailing padding, and decrease space around the search icon ([ref](https://github.com/woocommerce/woocommerce-ios/pull/8092#pullrequestreview-1176782138))
- Update bottom Continue CTA to use `safeAreaInset`
- Update SUGGESTIONS header text to footnote font
- Remove the skip button pecCkj-dp-p2
- Auto-dismiss the keyboard after selecting a domain and the continue button appears ([ref](https://github.com/woocommerce/woocommerce-ios/pull/8092#pullrequestreview-1176782138))

### Testing instructions


Please ignore the back button text in the navigation bar, this is updated in a separate PR https://github.com/woocommerce/woocommerce-ios/pull/8154.

- Log in if needed
- Go to the Menu tab, and tap `Switch store`
- On the store picker, tap `+ Add a store`
- Tap `Create a new store` --> the domain selector should be shown with a placeholder image
- Enter some store name and tap `Continue` --> the store name should be the initial query in the next domain search screen, and some domain results are loaded
- Remove the search query --> the empty state should match design and looks good in dark mode
- Enter some other search query --> the spinner should appear, and some search results are loaded shortly
- Tap to select a domain --> a `Continue` CTA should appear at the bottom, and the keyboard should be resigned
- Enter some very long query --> there should be some space in the search field trailing edge
- Replace the search query to be `Test` to have no results --> an error message should be shown, center-aligned

### Screenshots


empty query | suggestions | domain selected | loading | error
-- | -- | -- | -- | --
![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 25 44](https://user-images.githubusercontent.com/1945542/202980706-6302c6b0-7b1e-4267-88ac-7cd5d0c7137c.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 00](https://user-images.githubusercontent.com/1945542/202980719-1c813385-b203-4a75-aa89-753561d8c364.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 04](https://user-images.githubusercontent.com/1945542/202980732-eed4038f-f1e4-4b34-b94a-57b550481119.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 20](https://user-images.githubusercontent.com/1945542/202980741-e1cb3f31-3eb0-47e1-8634-0ddb900b7c24.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 15](https://user-images.githubusercontent.com/1945542/202980739-6300d3df-b03b-4c93-81dc-e9bd079d7f34.png)
![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 30](https://user-images.githubusercontent.com/1945542/202980743-9641f312-5487-4382-8c0c-575539d04e47.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 38](https://user-images.githubusercontent.com/1945542/202980746-b9103484-48be-4c8f-8c79-f03d37161b87.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 40](https://user-images.githubusercontent.com/1945542/202980749-a1d13243-268f-467f-9ac3-251fa4a858e3.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 26 48](https://user-images.githubusercontent.com/1945542/202980750-acd68c9c-e82b-421f-972a-0d33ed299105.png) | ![Simulator Screen Shot - iPhone 14 - 2022-11-21 at 14 27 19](https://user-images.githubusercontent.com/1945542/202980753-54588557-701d-470f-965f-ea256beddc12.png)


---
